### ISSUE TITLE
Certificate issued for HTTPS error page has validity period of more than two years

### ISSUE THREAD

**Issue by eblosug** [eblocker/eblocker#48]
How to reproduce this issue:

* Enable HTTPS on eBlocker
* Use CA certificate with validity period of three years (default)
* Verify that HTTPS works on pages with valid certificates
* Go to https://self-signed.badssl.com/
* Instead of showing an eBlocker error page, the browser will complain that the server certificate is invalid
* Let the browser show the certificate. Its validity period will have the same end date as eBlocker's CA certificate.

![Screen Shot 2020-08-18 at 11 02 11](https://user-images.githubusercontent.com/61415621/90494053-5dbdce00-e143-11ea-9042-4178d4537c94.png)

**Comment by eblosug** [comment]
In [AccessDeniedService.generateKeyPair](https://github.com/eblocker/eblocker/blob/045eb3ac42213875cf5ace6f809e6d6cfb85231d/eblocker-icapserver/src/main/java/org/eblocker/server/http/service/AccessDeniedService.java#L235) the maximum validity of 825 days must be taken into account.

**Comment by eblosug** [comment]
No, the certificate is created in [GeneratingKeyManager.generateCertificate()](https://github.com/eblocker/eblocker/blob/045eb3ac42213875cf5ace6f809e6d6cfb85231d/eblocker-icapserver/src/main/java/org/eblocker/server/common/ssl/GeneratingKeyManager.java#L125).

**Comment by eblosug** [comment]
PR #51 fixes the problem for pages that are blocked by parental controls.

But: temporary certificates for sites with invalid certificates, e.g. https://self-signed.badssl.com/ are issued by the Squid.

**Comment by kizm0** [comment]
is there a possibility to use a own certificate? in my environment all clients already trust my custom CA-certificate i would like to use this CA-certificate in eBlocker so no clients have to be adjusted...

**Comment by eblosug** [comment]
@kizm0 There is no way to import a CA via the user interface.

However, there is a [script](https://github.com/eblocker/eblocker/blob/develop/eblocker-icapserver/src/main/package/bin/import-ca-into-jks.sh) at

    /opt/eblocker-icap/bin/import-ca-into-jks.sh

that seems to be doing this. To be honest, I have never used it and don't know if it really works (it's six years old).

**Comment by eblosug** [comment]
The issue still occurs on Debian Bookworm with Squid 5.7.
